mongodb-advanced/components/people-component.js

```diff
@@ -6,25 +6,33 @@ export default {
   templateUrl: 'mongodb-advanced/people-component.html'
 };
 
-function Ctrl($scope, exPeopleService) {
+function Ctrl($scope, exPeopleService, brAlertService) {
   const self = this;
   self.showMyModal = false;
 
-  exPeopleService.getPeople(people => {
-    self.savedPeople = people;
-  });
+  self.getPeople = function() {
+    exPeopleService.getPeople(people => self.savedPeople = people)
+      .catch(err => {
+        brAlertService.add('error', err);
+      })
+      .catch(err => {
+        brAlertService.add('error', err);
+      });
+  };
+
+  self.getPeople();
 
   self.modalClosed = () => {
-    exPeopleService.getPeople(people => {
-      self.savedPeople = people;
-    });
+    self.getPeople();
   };
 
   self.deletePerson = name => {
-    exPeopleService.removePerson(name, () => {
-      exPeopleService.getPeople(people => {
-        self.savedPeople = people;
+    exPeopleService.removePerson(name)
+      .then(data => {
```

Comment:
if there is no data, or you're not interested in it...
```
.then(() => self.getPeople())
```
which is the same as
```
.then(() => {
  return self.getPeople();
})
```
there's another important difference with this syntax.  After making `getPeople` return a promise by the modifications described in my comment there, any `catch` or `then` in the promise chin will work as expected.

Since `getPeople` is catching it's own errors, this should actually be a promise chain like this.
```
exPeopleService.removePerson(name)
  .catch(err => brAllertService.add('error', err))
  .then(() => return self.getPeople());
  // no need to catch errors because `getPeople` does that already
```